tdigest.go

```diff
@@ -335,3 +345,96 @@ func (cs *centroidSet) MergeInto(other TDigest) {
 		other.Add(c.mean, c.count)
 	}
 }
+
+func (cs *centroidSet) MarshalBinary() ([]byte, error) {
+	buf := bytes.NewBuffer(nil)
+	w := &stickyErrWriter{w: buf, err: nil}
+	binary.Write(w, binary.LittleEndian, int32(1))
+
+	var min, max float64
+	if len(cs.centroids) > 0 {
+		min = cs.centroids[0].mean
+		max = cs.centroids[len(cs.centroids)-1].mean
+	}
+
+	binary.Write(w, binary.LittleEndian, min)
+	binary.Write(w, binary.LittleEndian, max)
+	binary.Write(w, binary.LittleEndian, float64(cs.compression))
+	binary.Write(w, binary.LittleEndian, int32(len(cs.centroids)))
+	if w.err != nil {
```

Comment:
Under what circumstances would this have an error? I think we know statically that there can't be any, so why not check only a single time (at the end)?

Reply:
Good point, removed this check.